src/borg/repository.py

```diff
@@ -182,9 +182,29 @@ def create(self, path):
 
     def save_config(self, path, config):
         config_path = os.path.join(path, 'config')
+        old_config_path = os.path.join(path, 'config.old')
+
+        if os.path.isfile(old_config_path):
+            logger.warning("Old config file not securely erased on previous config update")
+            self.secure_erase_config(old_config_path)
+
+        if os.path.isfile(config_path):
+            os.link(config_path, old_config_path)
```

Comment:
I think upgrade requires hardlinks, but that is not the regular usage case.

try/except checking for errno in (EMLINK, EPERM) should be sufficient, and can warn in these cases that the file couldn't be erased.